pkgs/development/compilers/dotnet/sdk/default.nix

```diff
@@ -0,0 +1,280 @@
+      leaveDotGit = true;
+    };
+    DOTNET_INSTALL_DIR = bootstrap-sdk;
+    # https://github.com/NixOS/nixpkgs/issues/38991
+    # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
+    LOCALE_ARCHIVE = lib.optionalString stdenv.isLinux
+        "${glibcLocales}/lib/locale/locale-archive";
+  }) ''
+    cp -Tr "$src" src
+    chmod -R +w src
+    cd src
+    git remote add origin https://github.com/dotnet/installer.git
+    patch -p1 < ${./0001-make-tarball-deterministic.patch}
+    jq '.tools=(.tools|.dotnet="6.0.107"|del(.runtimes))' global.json > global.json~
+    mv global.json~ global.json
+    patchShebangs *.sh
+    HOME=$(pwd)/fake-home \
+    DotNetBuildFromSource=true \
+      ./build.sh /p:ArcadeBuildTarball=true /p:TarballDir=\"$out\" /p:OfficialBuildId=20220101.1
+  '';
+
+in pkgs.clangStdenv.mkDerivation rec {
+  name = "dotnet-sdk";
+  version = "6.0.108";
+
+  src = dotnet-sdk-src;
+
+  nativeBuildInputs = [
+    curl
+    pkgs.git # used for patching
+    pkgs.python3
+    pkgs.cmake
+    pkgs.pkgconfig
+  ];
+
+  buildInputs = [
+    pkgs.llvm
+    pkgs.zlib
+    pkgs.icu
+  ] ++ lib.optional stdenv.isLinux pkgs.lttng-ust_2_12;
+
+  postUnpack = ''
+    # The build process tries to overwrite some things in the sdk (e.g.
+    # SourceBuild.MSBuildSdkResolver.dll), so it needs to be mutable.
+    cp -Tr ${bootstrap-sdk} $sourceRoot/.dotnet
+    chmod -R +w $sourceRoot/.dotnet
+    # Sets all timestamps to Jan 1 1980, the earliest mtime zips support.
+    find "$sourceRoot" -exec touch -t 198001010000.00 {} +
+  '';
+
+  postPatch = ''
+    # If this exists, --with-packages doesn't work.
+    rm -fr packages/archive
+    # these get copied around, but the empty objects/ directories get skipped,
+    # causing the repo to be invalid
+    for x in src/source-build/.git src/source-build/.git/modules/src/*; do touch $x/objects/.keep; done
+    patchShebangs $(find -name \*.sh -type f -executable)
+    # I'm not sure why this is required, but these files seem to use the wrong
+    # property name.
+    sed -i 's:\bVersionBase\b:VersionPrefix:g' \
+      src/clicommandlineparser/eng/Versions.props \
+      src/xliff-tasks/eng/Versions.props
+    # stop looking in /etc/os-release for distro rid
+    substituteInPlace \
+      ./src/runtime/eng/native/init-distro-rid.sh \
+      --replace '[ -e "''${rootfsDir}/etc/os-release" ]' 'false'
+    substituteInPlace \
+      src/runtime/src/libraries/Native/Unix/System.Net.Security.Native/pal_gssapi.c \
+      --replace '"libgssapi_krb5.so.2"' '"${pkgs.libkrb5}/lib/libgssapi_krb5.so.2"'
+    substituteInPlace \
+      src/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c \
+      --replace '"libicui18n.so"' '"${pkgs.icu}/lib/libicui18n.so"' \
+      --replace '"libicuuc.so"' '"${pkgs.icu}/lib/libicuuc.so"' \
+      --replace 'libicuucName[64]' 'libicuucName[256]' \
+      --replace 'libicui18nName[64]' 'libicui18nName[256]'
+  '';
```

Comment:
I meant the binaries that are generated by `dotnet build`, don't know if those get automagically fixed by the source build or not.

Since most `buildDotnetModule` packages actually run the binary it generates instead of calling `dotnet run` this is an important one as it'd make us not have to `patchelf` them any longer.

Reply:
> I meant the binaries that are generated by dotnet build, don't know if those get automatically fixed by the source build or not.

they should just be copies of e.g. `singlefilehost`, so that should be taken care of.